youtube_dl/extractor/funimation.py

```diff
@@ -110,11 +110,24 @@ def _search_kane(name):
             headers = {}
             if self._TOKEN:
                 headers['Authorization'] = 'Token %s' % self._TOKEN
-            sources = self._download_json(
-                'https://www.funimation.com/api/showexperience/%s/' % video_id,
-                video_id, headers=headers, query={
-                    'pinst_id': ''.join([random.choice(string.digits + string.ascii_letters) for _ in range(8)]),
-                })['items']
+            try:
+                sources = self._download_json(
+                    'https://www.funimation.com/api/showexperience/%s/' % video_id,
+                    video_id, headers=headers, query={
+                        'pinst_id': ''.join([random.choice(string.digits + string.ascii_letters) for _ in range(8)]),
+                    })
+                sources = sources['items']
+            except KeyError:
+                if 'errors' in sources:
```

Comment:
Read coding conventions.

Reply:
Understood. So in short, this extractor was written without "get()" - I'll try to clean it up a bit and push another commit.

Reply:
Before I commit, how does this look?

```python
        try:
            headers = {}
            if self._TOKEN:
                headers['Authorization'] = 'Token %s' % self._TOKEN

# + BEGIN EDITS +
            meta = self._download_json(
                'https://www.funimation.com/api/showexperience/%s/' % video_id,
                video_id, headers=headers, query={
                    'pinst_id': ''.join([random.choice(string.digits + string.ascii_letters) for _ in range(8)]),
                })
            sources = meta.get('sources')
            errors = meta.get('errors')
            if errors is not None:
                if len(errors) > 0:
                    error = errors[0]
                    detail = error.get('detail')
                    if detail is not None:
                        raise ExtractorError('%s said: %s' % (
                            self.IE_NAME, detail), expected=True)
                    else:
                        raise ExtractorError(error, expected=True)
# - END EDITS -

        except ExtractorError as e:
            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
                error = self._parse_json(e.cause.read(), video_id)['errors'][0]
                raise ExtractorError('%s said: %s' % (
                    self.IE_NAME, error.get('detail') or error.get('title')), expected=True)
            raise
```

Reply:
If errors is always a list and we want to iterate the whole list instead of just the first error:
```python3
        try:
            headers = {}
            if self._TOKEN:
                headers['Authorization'] = 'Token %s' % self._TOKEN

# + BEGIN NEW CODE +
            meta = self._download_json(
                'https://www.funimation.com/api/showexperience/%s/' % video_id,
                video_id, headers=headers, query={
                    'pinst_id': ''.join([random.choice(string.digits + string.ascii_letters) for _ in range(8)]),
                })
            sources = meta.get('items') or []
            errors = meta.get('errors')
            if errors:
                for error in errors:
                    detail = error.get('detail')
                    if detail:
                        raise ExtractorError('%s said: %s' % (
                            self.IE_NAME, detail), expected=True)
                    else:
                        raise ExtractorError(error, expected=True)
# - END NEW CODE -

        except ExtractorError as e:
            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
                error = self._parse_json(e.cause.read(), video_id)['errors'][0]
                raise ExtractorError('%s said: %s' % (
                    self.IE_NAME, error.get('detail') or error.get('title')), expected=True)
            raise
```

Reply:
More or less correct not taking code duplication into account.
>Aside from the fixes, should we loop and display all of the errors (instead of just `errors[0]`)?

You can if this make sense.
>Also, is `errors` always a list (remove the `isinstance` check)?

There is no such guarantee.

Reply:
Perfect. Yeah the code duplication was only there for reference of what is being changed. :) Submitting another commit now.